File: misc/demo/spring/src/main/java/com/dahuatech/spring/bean/Person.java
package com.dahuatech.spring.bean;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.validation.annotation.Validated;

import javax.annotation.PostConstruct;

/**
 * <p>projectName: demo</p>
 * <p>packageName: com.dahuatech.bean</p>
 * <p>className: Person</p>
 * <p>date: 2023/3/17</p>
 *
 * @author qinjiawei(alan)
 * @version 1.0.0
 * @since JDK8.0
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Component
public class Person {
    @Value("alan")
    private String name;
    @Value("28")
    private int age;

    @PostConstruct
    public void init() {
        System.out.println("constructor");
    }
}

File: parquet/src/main/scala/com/dahuatech/parquet/bean/Person.scala
package com.dahuatech.parquet.bean

import org.apache.avro.generic.GenericData

/**
 * <p>projectName: demo</p>
 * <p>packageName: com.dahuatech.parquet.bean</p>
 * <p>className: Person</p>
 * <p>date: 1/2/2024</p>
 *
 * @author qinjiawei(alan)
 * @since JDK8.0
 * @version 1.0.0
 */

@SerialVersionUID(234288747753438L)
case class Person(name: String, age: Long) {
  // 样例类/空参构造器必须这样设置 直接在类参数中设置通过反射调用空参构造器不生效
  def this() {
    this("", 0)
  }
}